includes/components/class-review.php

```diff
@@ -415,4 +444,23 @@ public function alter_review_view_block( $template ) {
 
 		return $template;
 	}
+
+	/**
+	 * Clear review cache.
+	 *
+	 * @param string     $new_status New comment status.
+	 * @param string     $old_status Old comment status.
+	 * @param WP_Comment $comment Comment object.
+	 */
+	public function clear_review_cache( $new_status, $old_status, $comment ) {
```

Comment:
it seems that you reset global comment cache every time one of the reviews gets another status, but it's better to store review count per listing and reset it without affecting all the other cached review counts for listings.